src/librustc_session/config.rs

```diff
@@ -442,6 +442,19 @@ impl Input {
             Input::Str { ref name, .. } => name.clone(),
         }
     }
+
+    pub fn is_rlink(&self) -> bool {
+        match *self {
+            Input::File(ref ifile) => {
+                if let Some(extension) = ifile.extension() {
+                    extension == RLINK_EXT
```

Comment:
This is technically a breaking change, as a source file may have `.rlink` as extension. Checking if the file starts with `{` or magic bytes (eg b"rlink", should be prepended before writing it) will fix this.

Reply:
I'm not sure if this is still necessary as the feature is now gated by `-Z link-only`.

Reply:
I think automatically detecting the `.rlink` file will be desired when we stabilize this feature; however, we will likely change the `.rlink` file to a binary at that time. So I prefer not to include automatic file format detection at this stage.